pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -117,30 +119,60 @@ func (r *Reconciler) create() error {
 		Items: metadataItems,
 	}
 
-	operation, err := r.computeService.InstancesInsert(r.projectID, zone, instance)
+	return instance, nil
+}
+
+// Create creates machine if and only if machine exists, handled by cluster-api
+func (r *Reconciler) create() error {
+	instance, err := r.newInstanceFromMachineScope()
 	if err != nil {
-		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
-			Type:    v1beta1.MachineCreated,
-			Reason:  machineCreationFailedReason,
-			Message: err.Error(),
-			Status:  corev1.ConditionFalse,
-		}); reconcileWithCloudError != nil {
-			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
-		}
-		return fmt.Errorf("failed to create instance via compute service: %v", err)
+		return err
 	}
-	if op, err := r.waitUntilOperationCompleted(zone, operation.Name); err != nil {
-		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
-			Type:    v1beta1.MachineCreated,
-			Reason:  machineCreationFailedReason,
-			Message: err.Error(),
-			Status:  corev1.ConditionFalse,
-		}); reconcileWithCloudError != nil {
-			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
-		}
-		return fmt.Errorf("failed to wait for create operation via compute service. Operation status: %v. Error: %v", op, err)
+
+	// Passing machine.UID makes this request and subsequent
+	// requests idempotent as long as _all_ the other parameters
+	// to InstancesInsert() are equal. Note: you cannot use the
+	// same UID for any other operation as it is now keyed to the
+	// insert operation.
+	operation, err := r.computeService.InstancesInsert(string(r.machine.UID), r.projectID, r.providerSpec.Zone, instance)
+	if err != nil {
+		return errors.Wrap(err, "InstancesInsert() failed")
 	}
-	return r.reconcileMachineWithCloudState(nil)
+
+	if operation.Status != operationDone {
+		klog.Infof("Create Operation #%v not %q for machine %q, returning an error to requeue", operation.Id, operationDone, r.machine.Name)
```

Comment:
when we not succeed we want to reflect it into the machine state/providerConditions  the likes of ```.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{		// Passing machine.UID makes this request and subsequent
			Type:    v1beta1.MachineCreated,		// requests idempotent as long as _all_ the other parameters
			Reason:  machineCreationFailedReason,		// to InstancesInsert() are equal. Note: you cannot use the
			Message: err.Error(),		// same UID for any other operation as it is now keyed to the
			Status:  corev1.ConditionFalse,		// insert operation.
		}); reconcileWithCloudError != nil {```
See a tested example of current approach https://github.com/openshift/cluster-api-provider-gcp/pull/14#issue-278378481

Reply:
Doesn't this only apply to `InstancesInsert()`? Polling `operation.Status==DONE` will yield no error.

Reply:
why wouldn't we want to reflect this "progressing" InstanceState on the machine status?